test-files/indentation-tests.cs

```diff
@@ -213,6 +213,18 @@ public class NestedPreProcessor
 #if DEBUG
     public static bool Debug = true;
 #endif
+
+    public void Foo()
+    {
+#if DEBUG
+        if (true)
+#else
+        if (true)
+#endif
+        {
```

Comment:
I think there are some indentation issue from line 221 to line 223.

Reply:
That sounds reasonable.

The csharp-mode indentation engine does not know (and cannot know) that semantically those 2 different if-statements represents the same lexical scope in two different build-configurations and should have their indentation data "compressed" into the same one.

To be honest, I'm not sure we can fix that without a complete rewrite.

Do you have any ideas?

Reply:
Not on top of my head right now. I will keep investigate and see if I can do anything with this...